pipe/pipe.go

```diff
@@ -208,3 +208,21 @@ type zeroReader struct{}
 func (zeroReader) Read([]byte) (int, error) {
 	return 0, io.EOF
 }
+
+// closeWriter if io.Writer happens to implement io.Closer interface
+func closeWriter(w io.Writer) error {
+	wc, ok := w.(io.Closer)
```

Comment:
Idea: close io.PipeReader and io.PipeWriter created by Run method internally only.